torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -484,7 +484,8 @@ def _compiled_autograd_key(ctx):
 
         @staticmethod
         def forward(ctx, *deduped_flat_tensor_args):
-            args = deduped_flat_tensor_args
```

Comment:
hmm, this is the `CompiledFunction.forward()`. I'm pretty sure that.. basically saved activations should never make it as inputs to this function? Since in the compiled autograd code path, the graph that compiled autograd sends to AOTAutograd is always an inference graph.

Basically I think I would have expected no change necessary here. But lmk if that's wrong!

Reply:
in inference mode, although compiled autograd's activations never make it in here, the pad_mm fx pass's tensors shapes do: https://fburl.com/1xdd7f2g. maybe the fix is instead to not set requires_grad on those tensors when we're in aot inference mode

Reply:
It would feel a bit better to fix this in inductor, yeah. Although... I'm not sure if that means we'll need to generate multiple variants of every pattern (a training and an inference variant)